src/services/dexes/uniswapV3/UniswapV3LiquidityResolver.ts

```diff
@@ -0,0 +1,131 @@
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+import {Token, ChainId} from '@uniswap/sdk-core';
+
+import TimeService from '../../TimeService.js';
+import {FeedsType} from '../../../types/Feed.js';
+import {FeedDataService} from '../../FeedDataService.js';
+import {LiquidityService} from './LiquidityService.js';
+import {LeavesAndFeeds} from '../../../types/Consensus.js';
+import {UniswapV3PoolRepository} from '../../../repositories/UniswapV3PoolRepository.js';
+import {ChainsIds} from '../../../types/ChainsIds.js';
+import {TokenRepository} from '../../../repositories/TokenRepository.js';
+import {UniswapV3Param} from './interfaces.js';
+import {DexProtocolName} from '../../../types/Dexes.js';
+import {UniswapV3Pool} from '../../../models/Pools.js';
+
+@injectable()
+export class UniswapV3LiquidityResolver {
+  @inject(FeedDataService) feedDataService!: FeedDataService;
+  @inject(TimeService) timeService!: TimeService;
+  @inject(TokenRepository) tokenRepository!: TokenRepository;
+  @inject(LiquidityService) liquidityService!: LiquidityService;
+  @inject(UniswapV3PoolRepository) uniswapV3PoolRepository!: UniswapV3PoolRepository;
+  @inject('Logger') logger!: Logger;
+
+  readonly protocol = DexProtocolName.UNISWAP_V3;
+  readonly logPrefix = '[UniswapV3LiquidityResolver]';
+
+  async apply(chainId: ChainsIds) {
+    const feeds = await this.getFeeds(chainId);
+
+    const uniswapV3Params = await this.feedDataService.getParamsByFetcherName<UniswapV3Param>(
+      feeds as LeavesAndFeeds,
+      'UniswapV3',
+    );
+
+    if (uniswapV3Params.length === 0) {
+      throw new Error(`${this.logPrefix}[${chainId}] No params for fetcher`);
+    }
+
+    await Promise.all(
+      uniswapV3Params.map(async (param) => {
+        const pools = await this.uniswapV3PoolRepository.find({
+          token0: param.token0,
+          token1: param.token1,
+          fromChain: [chainId],
+          protocol: this.protocol,
+        });
+
+        if (pools.length === 0) {
+          this.logger.error(`${this.logPrefix}[${chainId}] pool not found`);
+          return;
+        }
+
+        const [token0, token1] = await this.getTokens(param.token0, param.token1, chainId);
+        await this.getAndSaveLiquidities(pools, token0, token1, chainId);
+      }),
+    );
+  }
+
+  private async getFeeds(chainId: ChainsIds) {
+    const dataTimestamp = this.timeService.apply();
+    const data = await this.feedDataService.apply(dataTimestamp, FeedsType.CONSENSUS);
```

Comment:
why `FeedsType.CONSENSUS`?

Reply:
I need to pass one `FeedsType` for this apply. Should I use `DEVIATION_TRIGGER`? I used the one that sounds right, but I am not sure. Let me know

Reply:
I ask to make sure you understand what you are doing, please review the code so you know what is the valid approach.

Reply:
In my understanding it is the feed file that would be get. 
We have the Deviations feeds.yaml: https://github.com/umbrella-network/pegasus-feeds/blob/develop/dev/onChainDeviation128.8.yaml
or for the consensus: https://github.com/umbrella-network/pegasus-feeds/blob/develop/dev/bsc/feedsOnChain.yaml

Not sure if I got the idea right and what we should place here.